src/js_user.hpp

```diff
@@ -242,18 +242,23 @@ void UserClass<T>::link_credentials(ContextType ctx, ObjectType this_object, Arg
 
 template<typename T>
 void UserClass<T>::call_function(ContextType ctx, ObjectType this_object, Arguments& args, ReturnValue &) {
-    args.validate_count(3);
+    args.validate_count(4);
     auto user = get_internal<T, UserClass<T>>(ctx, this_object);
 
     auto name = Value::validated_to_string(ctx, args[0], "name");
     auto call_args_js = Value::validated_to_array(ctx, args[1], "args");
-    auto callback = Value::validated_to_function(ctx, args[2], "callback");
+    auto service = Value::is_undefined(ctx, args[2])
```

Comment:
if service is optional isn't it better to be last parameter. Or this is just because to follow JS convention where callback is last

Reply:
I generally try to follow the callback-last API principle since it makes calling code clearer. And in this case the only caller always explicitly provides all arguments, one of them is just frequently the value `undefined`.

Reply:
👍